docs-gb/kubernetes/resources/pipeline.md

```diff
@@ -10,6 +10,25 @@ The `steps` list shows three models: `tfsimple1`, `tfsimple2` and `tfsimple3`. T
 
 The output of the Pipeline is the output from the `tfsimple3` model.
 
+## Support for Cyclic Pipelines
+
+Seldon Core 2 supports cyclic pipelines, allowing feedback loops within the inference graph. However, this feature should be used with caution, as improper use may result in infinite loops and unexpected behavior.
+
+The risk of infinite loops arises from the way Kafka Streams performs stream joins. If a feedback message enters the pipeline within the join window interval, and reaches a step that already holds messages from a previous iteration, Kafka Streams may perform a join between messages from different iterations. This can lead to unintended message propagation and potentially an unbounded flow of messages through your Kafka topics.
```

Comment:
```suggestion
The risk of infinite loops stems from how Kafka Streams handles stream joins. If a feedback message re-enters the pipeline within the join window and reaches a step already holding messages from a previous iteration, Kafka Streams may join messages across iterations. This can trigger unintended message propagation, potentially resulting in an unbounded flow through your Kafka topics.
```